ansi/list.go

```diff
@@ -0,0 +1,282 @@
+
+	be := BlockElement{
+		Block:   &bytes.Buffer{},
+		Style:   cascadeStyle(bs.Current().Style, rules, false),
+		Margin:  true,
+		Newline: !e.IsNested,
+	}
+	bs.Push(be)
+
+	_, _ = renderText(w, bs.Parent().Style.StylePrimitive, rules.BlockPrefix)
+	_, _ = renderText(bs.Current().Block, bs.Current().Style.StylePrimitive, rules.Prefix)
+	return nil
+}
+
+// Finish finishes rendering a ListElement with hanging indent support.
+func (e *ListElement) Finish(w io.Writer, ctx RenderContext) error {
+	bs := ctx.blockStack
+	rules := bs.Current().Style
+
+	width := min(bs.Width(ctx), uint(math.MaxInt))
+	wrapWidth := int(width) //nolint:gosec
+	content := bs.Current().Block.String()
+
+	wrapped := wrapListContent(content, wrapWidth)
+
+	mw := NewMarginWriter(ctx, w, rules)
+	if _, err := io.WriteString(mw, wrapped); err != nil {
+		return fmt.Errorf("glamour: error writing to writer: %w", err)
+	}
+
+	if !e.IsNested {
+		if _, err := io.WriteString(mw, "\n"); err != nil {
+			return fmt.Errorf("glamour: error writing to writer: %w", err)
+		}
+	}
+
+	_, _ = renderText(w, bs.Current().Style.StylePrimitive, rules.Suffix)
+	_, _ = renderText(w, bs.Parent().Style.StylePrimitive, rules.BlockSuffix)
+
+	bs.Current().Block.Reset()
+	bs.Pop()
+	return nil
+}
+
+// listItemPrefix represents a detected list item prefix.
+type listItemPrefix struct {
+	isListItem bool
+	width      int
+	length     int
+}
+
+var numberedListRegex = regexp.MustCompile(`^(\d{1,3})\.\s`)
+
+// detectListItemPrefix detects if a line starts with a list item prefix.
+func detectListItemPrefix(plainLine string) listItemPrefix {
+	trimmed := strings.TrimLeft(plainLine, " ")
+	if trimmed == "" {
+		return listItemPrefix{isListItem: false}
+	}
+
+	bullets := []string{"• ", "◦ ", "▪ ", "▸ ", "‣ ", "⁃ ", "⁌ ", "⁍ "}
+	for _, bullet := range bullets {
+		if strings.HasPrefix(trimmed, bullet) {
+			return listItemPrefix{
+				isListItem: true,
+				width:      ansi.StringWidth(bullet),
+				length:     len(bullet),
+			}
+		}
+	}
+
+	taskPrefixes := []string{"[✓] ", "[ ] ", "[x] ", "[X] "}
+	for _, prefix := range taskPrefixes {
+		if strings.HasPrefix(trimmed, prefix) {
+			return listItemPrefix{
+				isListItem: true,
+				width:      ansi.StringWidth(prefix),
+				length:     len(prefix),
+			}
+		}
+	}
+
+	simpleTasks := []string{"✓ ", "✗ ", "☑ ", "☐ "}
+	for _, prefix := range simpleTasks {
+		if strings.HasPrefix(trimmed, prefix) {
+			return listItemPrefix{
+				isListItem: true,
+				width:      ansi.StringWidth(prefix),
+				length:     len(prefix),
+			}
+		}
+	}
+
+	if matches := numberedListRegex.FindStringSubmatch(trimmed); matches != nil {
+		fullMatch := matches[0]
+		return listItemPrefix{
+			isListItem: true,
+			width:      ansi.StringWidth(fullMatch),
+			length:     len(fullMatch),
+		}
+	}
+
+	return listItemPrefix{isListItem: false}
+}
+
+// wrapListContent wraps list content with proper hanging indentation.
+func wrapListContent(content string, wrapWidth int) string {
```

Comment:
Why don't you use ANSI aware wrapping logic from `charmbracelet/x/ansi` and `charmbracelet/lipgloss/v2`?